core/domain/exp_services.py

```diff
@@ -1710,6 +1710,516 @@ def is_voiceover_change_list(change_list):
+                    if original_state_name in new_to_old_state_names:
+                        original_state_name = new_to_old_state_names.pop(
+                            original_state_name)
+                    deleted_state_names.append(original_state_name)
+            elif change.cmd == exp_domain.CMD_RENAME_STATE:
+                old_state_name = change.old_state_name
+                new_state_name = change.new_state_name
+                if old_state_name in added_state_names:
+                    added_state_names.remove(old_state_name)
+                    added_state_names.append(new_state_name)
+                elif old_state_name in new_to_old_state_names:
+                    new_to_old_state_names[new_state_name] = (
+                        new_to_old_state_names.pop(old_state_name))
+                else:
+                    new_to_old_state_names[new_state_name] = old_state_name
+
+            elif change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY:
+                # A condition to store the name of the properties changed
+                # in changed_properties dict.
+                state_name = change.state_name
+                if state_name in new_to_old_state_names:
+                    state_name = new_to_old_state_names[change.state_name]
+                if state_name in changed_properties:
+                    if (change.property_name not in
+                            changed_properties[state_name]):
+                        changed_properties[state_name].append(
+                            change.property_name)
+                else:
+                    changed_properties[state_name] = [change.property_name]
+            elif change.cmd == exp_domain.CMD_ADD_WRITTEN_TRANSLATION:
+                changed_property = None
+                if change.content_id == 'content':
+                    changed_property = 'content'
+                elif change.content_id[:14] == 'ca_placeholder':
+                    changed_property = 'widget_customization_args'
+                elif change.content_id == 'default_outcome':
+                    changed_property = 'default_outcome'
+                elif change.content_id == 'solution':
+                    changed_property = 'solution'
+                elif change.content_id[:4] == 'hint':
+                    changed_property = 'hints'
+                elif change.content_id[:8] == 'feedback':
+                    changed_property = 'answer_groups'
+                elif change.content_id[:10] == 'rule_input':
+                    changed_property = 'answer_groups'
+                # A condition to store the name of the properties changed
+                # in changed_properties dict.
+                state_name = change.state_name
+                if state_name in new_to_old_state_names:
+                    state_name = new_to_old_state_names[change.state_name]
+                if state_name in changed_translations:
+                    if (changed_property not in
```

Comment:
The translation is only dependent on the content of the exploration which is indexed by content_id and the content_ids are unique throughout the exploration version. So, instead of relying on state properties we should check the changes in content. Is it possible to check the changes in content/contentIds for merging translation changes?

Note: This is important as we allow multiple translators to work on translating exploration in multiple languages.